script/libtest.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+# k8s helpers, specially "wait"-ers on pod ready/deleted/etc
+
+set -u
+
+T_BOLD=$(test -t 0 && tput bold)
+T_SGR0=$(test -t 0 && tput sgr0)
+
+typeset -i TOTAL_PASS=0
+typeset -i TOTAL_FAIL=0
+
+KUBECTL_BIN=$(which kubectl)
+KUBECFG_BIN=$(which kubecfg)
+
+# Wrapup kubectl, kubecfg for --context
+test_reset() {
+    TEST_CONTEXT=${1:?missing k8s context, e.g.: minikube}
+    TOTAL_PASS=0
+    TOTAL_FAIL=0
+}
+test_report() {
+    info "$* ${T_BOLD}PASS=$TOTAL_PASS FAIL=$TOTAL_FAIL${T_SGR0}"
+    return $TOTAL_FAIL
+}
+
+kubectl() {
+    ${KUBECTL_BIN:?} --context=${TEST_CONTEXT:?} "$@"
+}
+kubecfg() {
+    ${KUBECFG_BIN:?} --context=${TEST_CONTEXT:?} "$@"
+}
+
+## Generic helper functions
+info () {
+    echo "INFO: $@"
+}
+spin() {
+    local on_tty
+    test -t 1 && on_tty=true || on_tty=false
+    echo -n .; sleep $1; $on_tty && echo -ne "\r"; sleep $1
+}
+
+pass_or_fail() {
```

Comment:
maybe we should use a third party tool for the test framework? we have used https://github.com/sstephenson/bats in the past, what do you think?